courseraresearchexports/models/ExportRequest.py

```diff
@@ -0,0 +1,371 @@
+        if self.snapshot_at:
+            json_metadata['snapshotAt'] = datetime_to_ts(self.snapshot_at)
+
+        return json_metadata
+
+    @classmethod
+    def from_json(cls, json_metadata):
+        """
+        Deserialize ExportRequestMetaData from json object.
+        :param json_metadata:
+        :return export_request_metadata: ExportRequestMetadata
+        """
+        if json_metadata:
+            kwargs = {}
+            if json_metadata.get('createdBy'):
+                kwargs['created_by'] = json_metadata['createdBy']
+            if json_metadata.get('createdAt'):
+                kwargs['created_at'] = timestamp_to_dt(
+                    json_metadata['createdAt'])
+            if json_metadata.get('completedAt'):
+                kwargs['completed_at'] = timestamp_to_dt(
+                    json_metadata['completedAt'])
+            if json_metadata.get('startedAt'):
+                kwargs['started_at'] = timestamp_to_dt(
+                    json_metadata['startedAt'])
+            if json_metadata.get('snapshotAt'):
+                kwargs['snapshot_at'] = timestamp_to_dt(
+                    json_metadata['snapshotAt'])
+            return cls(**kwargs)
+
+        else:
+            return None
+
+
+class ExportRequestWithMetadata(ExportRequest):
+    """
+    Class representing a export request from Coursera's research data export
+    service with metadata about its status.
+    """
+
+    def __init__(self, course_id=None, partner_id=None, group_id=None,
+                 export_type=None, anonymity_level=None,
+                 statement_of_purpose=None, schema_names=None,
+                 interval=None, ignore_existing=None, id=None,
+                 status=None, download_link=None, metadata=None, **kwargs):
+        super(ExportRequestWithMetadata, self).__init__(
+            course_id=course_id, partner_id=partner_id,
+            group_id=group_id, export_type=export_type,
+            anonymity_level=anonymity_level,
+            statement_of_purpose=statement_of_purpose,
+            schema_names=schema_names, interval=interval,
+            ignore_existing=ignore_existing)
+        self.id = id
+        self.status = status
+        self.download_link = download_link
+        self.metadata = metadata
+
+    def to_json(self):
+        """
+        Serialize ExportRequestWithMetadata to json object
+        :return json_request:
+        """
+        json_request = ExportRequest.to_json(self)
+
+        if self.id:
+            json_request['id'] = self.id
+        if self.status:
+            json_request['status'] = self.status
+        if self.download_link:
+            json_request['downloadLink'] = self.download_link
+        if self.metadata:
+            json_request['metadata'] = self.metadata.to_json()
+
+        return json_request
+
+    @classmethod
+    def from_parent(cls, parent, id=None, status=None, download_link=None,
```

Comment:
It's the parent class `ExportRequest` in this case.  Should I just be explicit in this case? `from_export_request`?